### PR TITLE
fixes #3, added type to hotspot update call

### LINKED ISSUE #3: Cannot update hotspot from frontend
Hi!

First of all, love the project! Fantastic work!

I'm running into an issue with updating the hotspot settings from the frontend. The call returns a 400 Bad Request. The backend logs: `Bad body: json: unknown field "type"`.

This is due to the struct in the backend being:
```go
type hotspotUpdateOptions struct {
	Password *string `json:"password"`
	Ssid     string  `json:"ssid"`
}
```
The request that is send from the frontend is:
```json
{"type":"hotspot","password":"password","ssid":"ssid"}
```
I'd be happy to submit the fix myself, but I'm not sure if this needs to be fixed in the frontend by removing the type, or adding the type in the backend.

### PR DESCRIPTION
**Pull request by PretentiousPotatoPeeler**
I've tested it and it seems to work

Not too familiar with go so there might be some improvements possible :smile:

### CODE REVIEW

File: internal/api/network.go [hunk 1]
```diff
@@ -791,7 +791,7 @@ func (nm *NetworkManager) UpdateConnection(connection *Connection, name, ssid st
 	return nil
 }
 
-func (nm *NetworkManager) UpdateHotspot(hotspot *Hotspot, ssid string, password *string) error {
+func (nm *NetworkManager) UpdateHotspot(hotspot *Hotspot, ssid string, password *string, typeStr string) error {
```
Comment (kahlertl) [inline internal/api/network.go:794]: Since the type of the hotspot can never change, we do not need to pass it along the `UpdateHotspot` function.

**Review by kahlertl — CHANGES_REQUESTED** [review]
Thanks for the pull request :smiley: I think it can be even simpler by just allowing the `type` field in the JSON payload and just ignore it afterwards.

Besides this, can you reiterate on your commit message and be a little bit more verbose?

xkcd: git commit messages

Personally, I try to follow these rules https://chris.beams.io/posts/git-commit/

**Comment by kahlertl** [comment]
Btw, I you want to have a quick path forward, I can also fix the issue and create a release today?

**Comment by PretentiousPotatoPeeler** [comment]
Fair points! Also an interesting article about commit messages. I'll move some stuff around and submit a new pull request. 

You're of course free to fix the bug yourself, but I'd love to contribute to the project. I'll submit the new pull request today

**Comment by kahlertl** [comment]
No rush, I will wait for your contribution :smile: 

Btw, you can also salvage this pull request. No need to create a complete new one. I suggest that you just reopen this one once you have a new patch.

**Comment by PretentiousPotatoPeeler** [comment]
I've recreated the branch so the pull request can't be reopened. #5 is the new PR

**Pull request closed without merging**